source/Calamari/Kubernetes/Commands/KubernetesDiscoveryCommand.cs

```diff
@@ -142,24 +146,29 @@ bool TryGetDiscoveryContextJson(out string json)
             return (json = variables.Get(ContextVariableName)) != null;
         }
 
-        bool TryGetAuthenticationContextTypeAndDiscoveryContextScope(
-            string contextJson, 
-            out string type, 
-            out TargetDiscoveryScope scope)
+        bool TryGetAuthenticationContextTypeDiscoveryContextScopeAndAdditionalVariables(
+            string contextJson,
+            out string type,
+            out TargetDiscoveryScope scope,
+            out Dictionary<string,string> additionalVariables)
         {
             type = null;
             scope = null;
+            additionalVariables = null;
             try
             {
                 var discoveryContext = JsonConvert
                     .DeserializeObject<TargetDiscoveryContext<AuthenticationType>>(contextJson);
-                
+
                 type = discoveryContext?.Authentication?.Type;
                 scope = discoveryContext?.Scope;
-                
+                additionalVariables = discoveryContext?.AdditionalVariables != null
+                    ? new Dictionary<string, string>(discoveryContext.AdditionalVariables, StringComparer.OrdinalIgnoreCase)
+                    : new Dictionary<string, string>();
+
                 if (type != null && scope != null)
                     return true;
-                
+
```

Comment:
The error message logged in the `catch` block could be better.
Parsing could fail even `Type` or `Scope` can be extracted. E.g. when it includes an `AdditionalVariables` property that cannot be parsed as a dictionary.

Reply:
The above comment is redundant now that the `AdditionalVariables` property is gone, but the messages could still be improved to say `"Authentication Type"` instead of just `Type`.